plugins/woocommerce-blocks/tests/e2e/tests/templates/legacy-templates.block_theme.spec.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { test, expect } from '@woocommerce/e2e-playwright-utils';
+import { cli } from '@woocommerce/e2e-utils';
+
+test.describe( 'Legacy templates', async () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+	} );
+
+	test( 'woocommerce//* name is supported', async ( {
+		admin,
+		page,
+		editor,
+	} ) => {
+		const templateName = 'single-product';
+		const customText = 'This is a customized template.';
+
+		await test.step( 'Customize existing template to create DB entry', async () => {
+			await admin.visitSiteEditor( {
+				postId: `woocommerce/woocommerce//${ templateName }`,
+				postType: 'wp_template',
+				canvas: 'edit',
+			} );
+
+			const title = editor.canvas.getByText( 'Title' );
+
+			await title.click();
+			await title.press( 'Enter' );
+
+			const emptyBlock = editor.canvas
+				.getByLabel( 'Empty block' )
+				.first();
+
+			await emptyBlock.fill( customText );
+			await page.keyboard.press( 'Escape' );
+
+			await expect( editor.canvas.getByText( customText ) ).toBeVisible();
+
+			await editor.saveSiteEditorEntities();
+		} );
+
+		await test.step( 'Update created term to legacy format in the DB', async () => {
+			const cliOutput = await cli(
+				`npm run wp-env run tests-cli -- \
+					wp term update wp_theme woocommerce-woocommerce \
+						--by="slug" \
+						--name="woocommerce" \
+						--slug="woocommerce"`
+			);
+
+			expect( cliOutput.stdout ).toContain( 'Success: Term updated.' );
+		} );
+
+		await test.step( 'Verify the template can be loaded via legacy ID', async () => {
+			await admin.visitSiteEditor( {
+				postId: `woocommerce//${ templateName }`,
+				postType: 'wp_template',
+				canvas: 'edit',
+			} );
+
+			await expect( editor.canvas.getByText( customText ) ).toBeVisible();
+		} );
```

Comment:
What do you think about also verifying that:

1. The template is correctly applied in the frontend.
2. The template is accessible from the Site Editor UI (it should appear in the list of templates when going to Appearance > Editor).

Reply:
Added in https://github.com/woocommerce/woocommerce/pull/46269/commits/ee326b833cb6079b07159536bb349fb197d56fc4